### PR TITLE
Added OTLP tracer type

### PR DESCRIPTION
**Pull request by Rashmika998**

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2 [hunk 1]
```diff
@@ -1050,17 +1050,21 @@
...
+            <Enabled>{{apim.open_telemetry.remote_tracer.enable}}</Enabled>
+            <Name>{{apim.open_telemetry.remote_tracer.name}}</Name>
+            <Url>{{apim.open_telemetry.remote_tracer.url}}</Url>
+            <HostName>{{apim.open_telemetry.remote_tracer.hostname}}</HostName>
+            <Port>{{apim.open_telemetry.remote_tracer.port}}</Port>
             <Properties>
-                <HostName>{{apim.open_telemetry.remote_telemetry.properties.hostname}}</HostName>
-                <Port>{{apim.open_telemetry.remote_telemetry.properties.port}}</Port>
+                {% for property in apim.open_telemetry.remote_tracer.properties %}
+                <{{property.name}}>{{property.value}}</{{property.name}}>
```
Comment (tharindu1st) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2:1061]: `<Property name="{{property.name}}">{{property.value}}</Property>`

**Review by tharindu1st — CHANGES_REQUESTED** [review]

**Review by tharindu1st — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+        String headerProperty = getHeaderKeyProperty();
+        String endPointURL = configuration.getFirstProperty(TelemetryConstants.OTLP_CONFIG_URL) != null ?
+                configuration.getFirstProperty(TelemetryConstants.OTLP_CONFIG_URL) : null;
+        String headerKey = headerProperty.substring(TelemetryConstants.OPENTELEMETRY_PROPERTIES_PREFIX.length());
+
+        String headerValue = configuration.getFirstProperty(headerProperty) != null ?
+                configuration.getFirstProperty(headerProperty) : null;
+
+        if (endPointURL != null && headerValue != null) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java:63]: use StringUtils

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java (hunk 2 above)
Comment (Rashmika998) [reply components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java]: Added and resolved.

File: components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hsqldb.lib.StringUtil;
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java:35]: invalid import use commons.lang3

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java (hunk 3 above)
Comment (Rashmika998) [reply components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/telemetry/OTLPTelemetry.java:35]: Noted.

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**